javascript/api.ts

```diff
@@ -0,0 +1,17 @@
+
+export interface IConn {
+	read(len: number): Promise<Uint8Array|undefined>;
```

Comment:
is this `undefined` when you reach the EOF?

Reply:
i believe so